src/transformers/trainer.py

```diff
@@ -1651,7 +1651,8 @@ def _inner_training_loop(
         self.callback_handler.optimizer = self.optimizer
         self.callback_handler.lr_scheduler = self.lr_scheduler
         self.callback_handler.train_dataloader = train_dataloader
-        self.state.trial_name = self.hp_name(trial) if self.hp_name is not None else None
+        if self.hp_name is not None and (trial or self._trial) is not None:
```

Comment:
`trial` and `self._trial` cannot be different at this stage, so let's keep it at
```suggestion
        if self.hp_name is not None and trial is not None:
```

Reply:
Actually, it can be different.

See code here

https://github.com/huggingface/transformers/blob/bd469c40659ce76c81f69c7726759d249b4aef49/src/transformers/integrations.py#L172-L177

L175 set `self._trial`, but L177 `trial` parameter is None.

It happened when using DDP and Optuna or SigOpt together.

Reply:
Fixed.